test/extended/crdvalidation/apiserver.go

```diff
@@ -0,0 +1,45 @@
+package crdvalidation
+
+import (
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Suite:openshift/crdvalidation/apiserver] APIServer CR fields validation", func() {
+	var (
+		oc              = exutil.NewCLI("cluster-basic-auth", exutil.KubeConfigPath())
+		apiServerClient = oc.AdminConfigClient().ConfigV1().APIServers()
+	)
+	defer g.GinkgoRecover()
+
+	g.It("additionalCORSAllowedOrigins", func() {
+		apiServerOrig, err := apiServerClient.Get("cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer func() {
+			apiServerClient.Update(apiServerOrig)
+		}()
+
+		apiServer := apiServerOrig.DeepCopy()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		invalidRegexes := []string{
+			`(?!lookahead)`,
+			`(?<lookbehind)`,
+			`unfinished repeat{3`,
+			`no closing (parentheses`,
+			`unknownescapesequence\k`,
+		}
+
+		for _, re := range invalidRegexes {
+			apiServer.Spec.AdditionalCORSAllowedOrigins = []string{re}
+			_, err = apiServerClient.Update(apiServer)
+			o.Expect(err).To(o.HaveOccurred())
```

Comment:
if one of these updates succeed, the following will fail due to updated resource version. I would just use one invalid regex, no need to be exhaustive here. This is really just a smoke test.